src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -0,0 +1,170 @@
+
+        internal readonly string BaseType = typeof(AvailabilityData).Name;
+        internal readonly AvailabilityData Data;
+        private readonly TelemetryContext context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry()
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);
+            this.Data.testRunId = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry(string testName, DateTimeOffset testTimeStamp, TimeSpan duration, string runLocation, bool success, string message = null)
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);            
+            this.Data.testTimeStamp = testTimeStamp.ToString("o", CultureInfo.InvariantCulture);
+            this.Data.testName = testName;
+            this.Data.testRunId = Guid.NewGuid().ToString();
+            this.Data.duration = duration.ToString(string.Empty, CultureInfo.InvariantCulture);
+            this.Data.result = success ? TestResult.Pass : TestResult.Fail;
+            this.Data.runLocation = runLocation;
+            this.Data.message = message;           
+        }        
+
+        /// <summary>
+        /// Gets or sets the test run id guid
+        /// </summary>
+        public Guid Id
+        {
+            get { return Guid.Parse(this.Data.testRunId); }
+            set { this.Data.testRunId = value.ToString(); }
+        }
+
+        /// <summary>
+        /// Gets or sets date and time when the availability test was executed.
+        /// </summary>
+        public DateTimeOffset TestTimeStamp
+        {
+            get { return Utils.ValidateDateTimeOffset(this.Data.testTimeStamp); }
+            set { this.Data.testTimeStamp = value.ToString("o", CultureInfo.InvariantCulture); }
+        }
+
+        /// <summary>
+        /// Gets or sets the test name.
+        /// </summary>
+        public string TestName
+        {
+            get { return this.Data.testName; }
+            set { this.Data.testName = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets availability test duration.
+        /// </summary>
+        public TimeSpan Duration
+        {
+            get
+            {
+                //TODO: Note: this ends up logging an error messsage for Request Telemetry if it fails, Both PageView and Request telemetry use this to validate so PV already has the bug.
+                return Utils.ValidateDuration(this.Data.duration);
+            }
+            set
+            {
+                this.Data.duration = value.ToString();
```

Comment:
Why is it validated on Get instead of on Set?  It seems like what is set should be validated so that get does not need to validate because only valid data can be set.